spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/SSLConnectionTests.java

```diff
@@ -44,4 +49,27 @@ public void test() throws Exception {
 		conn.close();
 	}
 
+	@Test
+	public void testAlgNoProps() throws Exception {
+		RabbitConnectionFactoryBean fb = new RabbitConnectionFactoryBean();
+		ConnectionFactory rabbitCf = spy(TestUtils.getPropertyValue(fb, "connectionFactory", ConnectionFactory.class));
+		new DirectFieldAccessor(fb).setPropertyValue("connectionFactory", rabbitCf);
+		fb.setUseSSL(true);
+		fb.setSslAlgorithm("TLSv1.2");
+		fb.afterPropertiesSet();
+		fb.getObject();
+		verify(rabbitCf).useSslProtocol("TLSv1.2");
+	}
+
+	@Test
+	public void testNoAlgNoProps() throws Exception {
+		RabbitConnectionFactoryBean fb = new RabbitConnectionFactoryBean();
+		ConnectionFactory rabbitCf = spy(TestUtils.getPropertyValue(fb, "connectionFactory", ConnectionFactory.class));
+		new DirectFieldAccessor(fb).setPropertyValue("connectionFactory", rabbitCf);
+		fb.setUseSSL(true);
+		fb.afterPropertiesSet();
+		fb.getObject();
+		verify(rabbitCf).useSslProtocol();
```

Comment:
Nothing to complain about, but before I merge it, please, let me know if we should change our default algorithm from `TLSv1.1` to `TLSv1` as it is in the `com.rabbitmq.client.ConnectionFactory`.

Or have I missed anything and it was really intentional to have it like this?